### PR TITLE
AST-1630 - WooCommerce Product layout column breaks

### PR DESCRIPTION
**Pull request by GauravKhupse**
### Description
With the latest Theme update v3.7.9, the WooCommerce shortcode for [product] layout breaks down.

Discussion forum: [WooCommerce shortcode issue with Astra last version (3.7.9)] https://wordpress.org/support/topic/woocommerce-shortcode-issue-with-astra-last-version-3-7-9/

### Types of changes

Breaking change 

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/compatibility/woocommerce/class-astra-woocommerce.php [hunk 1]
```diff
@@ -1278,7 +1278,7 @@ public function add_styles() {
 			);
 
 			if ( Astra_Builder_Helper::apply_flex_based_css() ) {
-				$css_global_button_tablet['.woocommerce ul.products, .woocommerce-page ul.products'] = array(
+				$css_global_button_tablet['.woocommerce ul.products:not(.elementor-grid), .woocommerce-page ul.products'] = array(
```
Comment (premanshup) [inline inc/compatibility/woocommerce/class-astra-woocommerce.php:1281]: @GauravKhupse Any reason we are making changes to existing CSS. Please confirm this with @nileshc-bsf and @imnavanath .

**Review by premanshup — CHANGES_REQUESTED** [review]

File: inc/compatibility/woocommerce/class-astra-woocommerce.php (hunk 1 above)
Comment (GauravKhupse) [reply inc/compatibility/woocommerce/class-astra-woocommerce.php]: shubham has added CSS only for desktop but he has not made changes for the responsive devices so the column setting is not working for responsive devices found as per @Swapnildhanrale We have fixed the issue for responsive devices.

File: inc/compatibility/woocommerce/class-astra-woocommerce.php [hunk 2]
```diff
@@ -1111,7 +1111,7 @@ public function add_styles() {
 				$archive_tablet_grid = $this->get_grid_column_count( 'archive', 'tablet' );
 
 				$tablet_css_shop_page_grid = array(
-					'.woocommerce.tablet-columns-' . $archive_tablet_grid . ' ul.products li.product, .woocommerce-page.tablet-columns-' . $archive_tablet_grid . ' ul.products' => array(
+					'.woocommerce.tablet-columns-' . $archive_tablet_grid . ' ul.products li.product:not(.elementor-grid), .woocommerce-page.tablet-columns-' . $archive_tablet_grid . ' ul.products:not(.elementor-grid)' => array(
```
Comment (imnavanath) [inline inc/compatibility/woocommerce/class-astra-woocommerce.php:1114]: li.product don't have .elementor-grid class its only applying for ul.products. Please do the needful.

Comment (nileshc-bsf) [reply inc/compatibility/woocommerce/class-astra-woocommerce.php:1114]: @imnavanath - Correct, it should for `ul.products` not for the `li.product`

Comment (GauravKhupse) [reply inc/compatibility/woocommerce/class-astra-woocommerce.php]: Changes done!

File: inc/compatibility/woocommerce/class-astra-woocommerce.php [hunk 3]
```diff
@@ -1393,10 +1390,10 @@ public function add_styles() {
 				$archive_mobile_grid = $this->get_grid_column_count( 'archive', 'mobile' );
 				$single_mobile_grid  = $this->get_grid_column_count( 'single', 'mobile' );
 
-				$css_global_button_mobile[ '.woocommerce ul.products, .woocommerce-page ul.products, .woocommerce.mobile-columns-' . $archive_mobile_grid . ' ul.products, .woocommerce-page.mobile-columns-' . $archive_mobile_grid . ' ul.products' ] = array(
+				$css_global_button_mobile[ '.ast-container .woocommerce ul.products:not(.elementor-grid), .woocommerce-page ul.products:not(.elementor-grid), .woocommerce.mobile-columns-' . $archive_mobile_grid . ' ul.products:not(.elementor-grid), .woocommerce-page.mobile-columns-' . $archive_mobile_grid . ' ul.products:not(.elementor-grid)' ] = array(
 					'grid-template-columns' => 'repeat(' . $archive_mobile_grid . ', minmax(0, 1fr))',
 				);
-				$css_global_button_mobile[ '.woocommerce.mobile-rel-up-columns-' . $single_mobile_grid . ' ul.products' ] = array(
+				$css_global_button_mobile[ '.ast-container .woocommerce.mobile-rel-up-columns-' . $single_mobile_grid . ' ul.products::not(.elementor-grid)' ] = array(
```
Comment (imnavanath) [inline inc/compatibility/woocommerce/class-astra-woocommerce.php:1396]: mobile-rel-up-columns- class comes for body, then how `.ast-container` comes before that in CSS?

Comment (GauravKhupse) [reply inc/compatibility/woocommerce/class-astra-woocommerce.php]: Removed, selector changes done.

Comment (GauravKhupse) [reply inc/compatibility/woocommerce/class-astra-woocommerce.php]: updated selector, changes done!

File: inc/compatibility/woocommerce/class-astra-woocommerce.php [hunk 4]
```diff
@@ -1393,10 +1390,10 @@ public function add_styles() {
 				$archive_mobile_grid = $this->get_grid_column_count( 'archive', 'mobile' );
 				$single_mobile_grid  = $this->get_grid_column_count( 'single', 'mobile' );
 
-				$css_global_button_mobile[ '.woocommerce ul.products, .woocommerce-page ul.products, .woocommerce.mobile-columns-' . $archive_mobile_grid . ' ul.products, .woocommerce-page.mobile-columns-' . $archive_mobile_grid . ' ul.products' ] = array(
+				$css_global_button_mobile[ '.ast-container .woocommerce ul.products:not(.elementor-grid), .woocommerce-page ul.products:not(.elementor-grid), .woocommerce.mobile-columns-' . $archive_mobile_grid . ' ul.products:not(.elementor-grid), .woocommerce-page.mobile-columns-' . $archive_mobile_grid . ' ul.products:not(.elementor-grid)' ] = array(
 					'grid-template-columns' => 'repeat(' . $archive_mobile_grid . ', minmax(0, 1fr))',
 				);
-				$css_global_button_mobile[ '.woocommerce.mobile-rel-up-columns-' . $single_mobile_grid . ' ul.products' ] = array(
+				$css_global_button_mobile[ '.woocommerce.mobile-rel-up-columns-' . $single_mobile_grid . ' ul.products::not(.elementor-grid)' ] = array(
```
Comment (imnavanath) [inline inc/compatibility/woocommerce/class-astra-woocommerce.php:1396]: This will work right as we removed one selector from here?

Comment (GauravKhupse) [reply inc/compatibility/woocommerce/class-astra-woocommerce.php:1396]: yes!

File: inc/compatibility/woocommerce/class-astra-woocommerce.php [hunk 5]
```diff
@@ -1111,7 +1111,7 @@ public function add_styles() {
 				$archive_tablet_grid = $this->get_grid_column_count( 'archive', 'tablet' );
 
 				$tablet_css_shop_page_grid = array(
-					'.woocommerce.tablet-columns-' . $archive_tablet_grid . ' ul.products li.product, .woocommerce-page.tablet-columns-' . $archive_tablet_grid . ' ul.products' => array(
+					'.woocommerce.tablet-columns-' . $archive_tablet_grid . ' ul.products li.product, .woocommerce-page.tablet-columns-' . $archive_tablet_grid . ' ul.products:not(.elementor-grid)' => array(
```
Comment (imnavanath) [inline inc/compatibility/woocommerce/class-astra-woocommerce.php:1114]: here one ul.products is without elementor-grid, that should also updated, correct me if I'm wrong.

**Review by imnavanath — APPROVED** [review]

File: inc/compatibility/woocommerce/class-astra-woocommerce.php (hunk 5 above)
Comment (GauravKhupse) [reply inc/compatibility/woocommerce/class-astra-woocommerce.php:1114]: Currently, we have provided CSS for ul, so it might be not needed for this

**Review by nileshc-bsf — APPROVED** [review]

**Review by premanshup — APPROVED** [review]

**Pull request merged**